test/public/stageView.spec.ts

```diff
@@ -0,0 +1,98 @@
+import { _uninitialize, _initialize } from '../../src/public/publicAPIs';
+import { Utils } from '../utils';
+import { stageView } from '../../src/public';
+
+describe('stageView', () => {
+  const utils = new Utils();
+
+  beforeEach(() => {
+    utils.processMessage = null;
+    utils.messages = [];
+    utils.childMessages = [];
+    utils.childWindow.closed = false;
+
+    // Set a mock window for testing
+    _initialize(utils.mockWindow);
+  });
+
+  afterEach(() => {
+    // Reset the object since it's a singleton
+    if (_uninitialize) {
+      _uninitialize();
+    }
+  });
+
+  const stageViewParams: stageView.StageViewParams = {
+    appId: 'appId',
+    contentUrl: 'contentUrl',
+    threadId: 'threadId',
+    title: 'title',
+    websiteUrl: 'websiteUrl',
+    entityId: 'entityId',
+  };
+
+  describe('open', () => {
+    it('should not allow calls before initialization', () => {
+      expect(() => stageView.open(stageViewParams)).toThrowError('The library has not yet been initialized');
+    });
+
+    test.each([['settings'], ['sidePanel'], ['authentication'], ['remove'], ['task'], ['meetingStage'], ['stage']])(
```

Comment:
How would you feel about modifying this slightly to be more like:

 Object.keys(FrameContexts)
      .map(k => FrameContexts[k])
      .forEach(frameContext => {
        it(`should not allow calls from ${frameContext} context`, async () => {

The reason I like this better is that it will catch issues in the future in case FrameContexts change or new ones are added.

Reply:
(Or you might be able to replace your inner array with something like tests.each([Object.keys(FrameContexts)]))